### PR TITLE
Add support for SageMaker Studio Analytics Extension

### PR DESCRIPTION
**Pull request by navinsoni**
*Issue #, if available:*

*Description of changes:*
This PR adds SageMaker Studio Analytics Extension to SageMaker Distribution image.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/cpu.env.in [hunk 1]
```diff
@@ -20,3 +20,7 @@ conda-forge::boto3[version='>=1.28.35,']
 conda-forge::sagemaker-python-sdk[version='>=2.173.0,']
 conda-forge::supervisor[version='>=4,']
 conda-forge::autogluon[version='>=0.8.2,']
+conda-forge::sagemaker-studio-sparkmagic-lib[version='>0.1.2,<1']
```
Comment (just4brown) [inline build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/cpu.env.in:23]: Both `sagemaker-studio-sparkmagic-lib` and `sparkmagic` are direct downstream dependencies of `sagemaker-studio-analytics-extension` correct?

I suggest removing them from the env file, instead allowing the conda resolver to bring in the correct versions as dictated by the constraints of `sagemaker-studio-analytics-extension`

File: build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/Dockerfile [hunk 2]
```diff
@@ -24,11 +25,13 @@ RUN usermod "--login=${NB_USER}" "--home=/home/${NB_USER}" --move-home "-u ${NB_
 ENV MAMBA_USER=$NB_USER
 
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends sudo gettext-base wget curl unzip git rsync && \
+    apt-get install -y --no-install-recommends sudo gettext-base wget curl unzip git rsync build-essential && \
     # We just install tzdata below but leave default time zone as UTC. This helps packages like Pandas to function correctly.
     DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends awscli tzdata && \
     chmod g+w /etc/passwd && \
     echo "ALL    ALL=(ALL)    NOPASSWD:    ALL" >> /etc/sudoers && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends krb5-user && \
```
Comment (just4brown) [inline build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/Dockerfile:33]: Is there any reason we can't append `krb5-user` to the previous installation line where awscli and tzdata are installed?

File: build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/Dockerfile [hunk 3]
```diff
@@ -80,4 +85,16 @@ RUN mkdir -p /var/run/supervisord && \
 USER $MAMBA_USER
 ENV PATH="/opt/conda/bin:/opt/conda/condabin:$PATH"
 WORKDIR "/home/${NB_USER}"
+
+# https://stackoverflow.com/questions/122327
+RUN SYSTEM_PYTHON_PATH=$(python3 -c "from __future__ import print_function;import sysconfig; print(sysconfig.get_paths().get('purelib'))") && \
+    # Move Spark and PySpark Kernel to conda env so that it appears in JupyterLab
+    #cp -a ${SYSTEM_PYTHON_PATH}/sparkmagic/kernels/sparkkernel /opt/conda/share/jupyter/kernels/ && \
```
Comment (just4brown) [inline build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/Dockerfile:92]: Can we remove the commented-out `cp` commands?

**Comment by just4brown** [comment]
Please add a `*.test.Dockerfile` for this package here: https://github.com/aws/sagemaker-distribution/tree/1.0.0-beta/test/test_artifacts/v1

Granted, from prior conversations we've discussed that we'll be limited in what we can test because of infrastructure/network constraints. But I think it was proposed that magic commands and the like can be exercised to validate basic functionality

**Review by just4brown — CHANGES_REQUESTED** [review]

**Comment by claytonparnell** [comment]
We need to update the [template Dockerfile](https://github.com/aws/sagemaker-distribution/blob/1.0.0-beta/template/Dockerfile) with same changes as well

File: build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/Dockerfile (hunk 2 above)
Comment (navinsoni) [reply build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/Dockerfile:33]: I wanted to keep it separate as it's doing a specific thing of adding a Kerberos user and setting up permissions for it.

File: build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/Dockerfile (hunk 3 above)
Comment (navinsoni) [reply build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/Dockerfile:92]: Yup will do it in next commit 👍

File: build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/cpu.env.in (hunk 1 above)
Comment (navinsoni) [reply build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/cpu.env.in:23]: Sure let me check and remove dependencies accordingly

File: build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/Dockerfile (hunk 2 above)
Comment (navinsoni) [reply build_artifacts/v1/v1.0/v1.0.0/v1.0.0-beta/Dockerfile:33]: Sure I can do that it's just easier to manage if we want to remove dependency.

**Review by claytonparnell — APPROVED** [review]

**Review by just4brown — APPROVED** [review]

**Pull request merged**